libraries/AP_HAL_ChibiOS/hwdef/F4BY_H743/hwdef.dat

```diff
@@ -0,0 +1,189 @@
+
+
+SERIAL_ORDER OTG1 USART2 USART1 USART3 UART4 UART5 
+DMA_PRIORITY S*
+
+
+# CAN bus
+PD0 CAN1_RX CAN1
+PD1 CAN1_TX CAN1
+
+# UART1 
+PB6 USART1_TX USART1 # TIM4_CH1 TIM16_CH1N
+PB7 USART1_RX USART1 # TIM4_CH2 TIM17_CH1N 
+
+# USART2 
+PD5 USART2_TX USART2 NODMA
+PD6 USART2_RX USART2
+
+# USART3 
+PD8 USART3_TX USART3 NODMA
+PD9 USART3_RX USART3
+
+PC10 UART4_TX UART4 NODMA
+PC11 UART4_RX UART4
+
+# SHARE dma with I2C2_TX
+PC12 UART5_TX UART5 NODMA
+PD2  UART5_RX UART5 NODMA
+
+define HAL_STORAGE_SIZE 16384
+
+#SPI1 for MPU
+PA5 SPI1_SCK SPI1
+PA6 SPI1_MISO SPI1
+PA7 SPI1_MOSI SPI1        # for board version 3.0.3
+# PD7 SPI1_MOSI SPI1 ALT(1) # for board version 3.0.4 and above
+PA4 MPU_CS CS
+ 
+# spi bus for dataflash AND SD
+PB13 SPI2_SCK SPI2
+PB14 SPI2_MISO SPI2
+PB15 SPI2_MOSI SPI2
+
+#SPI for iim
+PB3 SPI3_SCK SPI3
+PB4 SPI3_MISO SPI3
+PB5 SPI3_MOSI SPI3
+PA15 IIM_CS CS 
+
+PB12 FRAM_CS CS SPEED_VERYLOW
+PE15 FLASH_CS CS
+
+SPIDEV imu2     SPI3 DEVID1 IIM_CS  MODE3 2*MHZ  16*MHZ
+SPIDEV imu1     SPI1 DEVID1 MPU_CS  MODE3 2*MHZ  16*MHZ
+SPIDEV mpu6000  SPI1 DEVID1 MPU_CS  MODE3 1*MHZ   8*MHZ
+SPIDEV ramtron SPI2 DEVID1 FRAM_CS  MODE3 8*MHZ   8*MHZ
+SPIDEV sdcard  SPI2 DEVID2 FLASH_CS MODE0 400*KHZ 25*MHZ
+
+#  IMU mpu6000 or newer Invensense IMUs in main board
+IMU Invensense SPI:mpu6000 ROTATION_NONE 
+IMU Invensensev3 SPI:imu2 ROTATION_ROLL_180
+
+# also support external newer Invensense IMUs
+IMU Invensensev3 SPI:imu1 ROTATION_ROLL_180
+
+define HAL_INS_HIGHRES_SAMPLE 7
+define HAL_DEFAULT_INS_FAST_SAMPLE 7
+
+
+# one baro
+define AP_BARO_MS5611_ENABLED 1
+BARO MS5611 I2C:0:0x77
+
+# enable RAMTROM parameter storage
+define HAL_WITH_RAMTRON 1
+# enable FAT filesystem support
+define HAL_OS_FATFS_IO 1
+
+# this defines the default maximum clock on I2C devices.
+define HAL_I2C_MAX_CLOCK 100000
+I2C_ORDER I2C2 I2C1
+PB8 I2C1_SCL I2C1
+PB9 I2C1_SDA I2C1
+PB10 I2C2_SCL I2C2
+PB11 I2C2_SDA I2C2
+
+# look for I2C compass
+COMPASS HMC5843 I2C:0:0x1E false ROTATION_YAW_270
+COMPASS QMC5883L I2C:0:0x0D false ROTATION_YAW_180
+COMPASS QMC5883P I2C:0:0x2C false ROTATION_YAW_180
+define HAL_PROBE_EXTERNAL_I2C_COMPASSES
+
+# PWM out pins
+PA0 TIM2_CH1 TIM2 PWM(1) GPIO(50)
+PA1 TIM2_CH2 TIM2 PWM(2) GPIO(51)
+PA2 TIM2_CH3 TIM2 PWM(3) GPIO(52)
+PA3 TIM2_CH4 TIM2 PWM(4) GPIO(53)
+
+PE9  TIM1_CH1 TIM1 PWM(5) GPIO(54)
+PE11 TIM1_CH2 TIM1 PWM(6) GPIO(55)
+PE13 TIM1_CH3 TIM1 PWM(7) GPIO(56)
+PE14 TIM1_CH4 TIM1 PWM(8) GPIO(57)
+
+PD13 TIM4_CH2 TIM4 PWM(9) GPIO(58)
+PD12 TIM4_CH1 TIM4 PWM(10) GPIO(59)
+PD15 TIM4_CH4 TIM4 PWM(11) GPIO(60)
+PD14 TIM4_CH3 TIM4 PWM(12) GPIO(61)
+
+PC7 TIM8_CH2 TIM8 PWM(13) GPIO(62)
+PC6 TIM8_CH1 TIM8 PWM(14) GPIO(63)
+PC8 TIM8_CH3 TIM8 PWM(15) GPIO(64)
+PC9 TIM8_CH4 TIM8 PWM(16) GPIO(65) 
+
+# classic rcin for protocols include PPM
+PB0 TIM3_CH3 TIM3  RCININT PULLDOWN LOW
+
+# use serial2 for RCIN by default for all serial rc protocols
+define DEFAULT_SERIAL2_PROTOCOL SerialProtocol_RCIN
+define DEFAULT_SERIAL2_BAUD 460
```

Comment:
remove, not required... RCin protocol automatically sets baud rate as it auto configures